qiita_pet/handlers/analysis_handlers.py

```diff
@@ -154,21 +154,54 @@ def get(self, analysis_id):
                 dropped[data_type].append((Study(study).title, len(samples),
                                            ', '.join(samples)))
 
-        self.render("analysis_results.html",
+        self.render("analysis_results.html", analysis_id=analysis_id,
                     jobres=jobres, aname=analysis.name, dropped=dropped,
                     basefolder=get_db_files_base_dir())
 
+    @authenticated
+    def post(self, analysis_id):
+        analysis_id = int(analysis_id.split("/")[0])
+        analysis_id_sent = int(self.get_argument('analysis_id'))
+        action = self.get_argument('action')
+
+        if analysis_id != analysis_id_sent or action != 'delete_analysis':
+            raise QiitaPetAuthorizationError(
+                self.current_user.id,
+                'analysis/results/%d-delete' % analysis_id)
+
+        analysis = Analysis(analysis_id)
+        analysis_name = analysis.name
+        check_analysis_access(self.current_user, analysis)
+
+        try:
+            Analysis.delete(analysis_id)
+            msg = ("Analysis <b><i>%s</i></b> has been deleted." % (
+                analysis_name))
+            msg_level = "success"
+        except Exception as e:
+            msg = ("Couldn't remove <b><i>%s</i></b> analysis: %s" % (
+                analysis_name, str(e)))
+            msg_level = "danger"
+
+        # redirecting to list of analysis but also passing messages and
+        # we need to change the request.method to GET
+        self.request.method = 'GET'
+        ShowAnalysesHandler(self.application, self.request)._execute(
```

Comment:
I tested that and it didn't work. Simply try: http://qiita.ucsd.edu/?msg=Jose&msg_level=danger